src/com/xilinx/rapidwright/timing/TimingGraph.java

```diff
@@ -1602,53 +1595,42 @@ else if (mycellname.startsWith("LUT") || mycellname.startsWith("RAM") || mycelln
                 }
 
             }
-            else if (mycellname.startsWith("CARRY")) {
-                List<String> config = new ArrayList<String>();
-
-
-                Cell c = design.getCell(k);
-                if (c == null) {
-                    continue;
-                }
-
+            else if (myCellName.startsWith("CARRY")) {
+                int encodedConfig = 0;
                 if (c.getPhysicalPinMapping("CI") == null) {
-                    config.add("CYINIT_BOT:GND");
-                    config.add("CARRY_TYPE:SINGLE_CY8");
-                }
-                else if (c.getPhysicalPinMapping("CI_TOP") == null) {
-                    config.add("CYINIT_TOP:GND");
-                    config.add("CARRY_TYPE:SINGLE_CY8");
+                	encodedConfig |= intrasiteAndLogicDelayModel.getEncodedConfigCode("CARRY8:CYINIT_BOT:GND");
+                } else if (c.getPhysicalPinMapping("CI_TOP") == null) {
+                	encodedConfig |= intrasiteAndLogicDelayModel.getEncodedConfigCode("CARRY8:CYINIT_TOP:GND");
                 } else {
-                    config.add("CYINIT_BOT:CIN"); config.add("CARRY_TYPE:SINGLE_CY8");
+                	encodedConfig |= intrasiteAndLogicDelayModel.getEncodedConfigCode("CARRY8:CYINIT_BOT:CIN"); 
                 }
-
+            	encodedConfig |= intrasiteAndLogicDelayModel.getEncodedConfigCode("CARRY8:CARRY_TYPE:SINGLE_CY8");                
+            	short belIdx = intrasiteAndLogicDelayModel.getBELIndex("CARRY8");
+            	
                 for (EDIFPortInst ep1 : portInstList) {
                     if (!ep1.isInput()) {
                         continue;
                     }
 
-                    String s1 = k + "/" + ep1.getName();
+                    String s1 = cellName + "/" + ep1.getName();
                     for (EDIFPortInst ep2 : portInstList) {
 
                         if (!ep2.isOutput()) {
                             continue;
                         }
-                        String s2 = k + "/" + ep2.getName();
+                        String s2 = cellName + "/" + ep2.getName();
                         float logicDelay = 0.0f;
                         if (ep1 != ep2 && ep1.isInput() && ep2.isOutput()) {
-
-                            BEL mybel = c.getSiteInst().getBEL(mycellType.toString());
-                            c = c.getSiteInst().getCell(mybel);
-
                             String physPin = c.getPhysicalPinMapping(ep1.getName());
                             String outputPhysPin = c.getPhysicalPinMapping(ep2.getName());
 
                             if (physPin == null || physPin.equals("null")) {
-                                config = new ArrayList<>();
+                            	// TODO - This is suspected to be buggy behavior

```

Comment:
@YunxZhou - This code does not seem to be correct.  It will clear out the config parameters for the cell once it finds a pin that is null.  I couldn't remove this behavior as it would change the routing result, but we should investigate why this code is here.

@pongstorn-xilinx  - FYI